api/cpp/include/slint_string.h

```diff
@@ -127,6 +127,34 @@ struct SharedString
     /// \endcode
     static SharedString from_number(double n) { return SharedString(n); }
 
+    /// Returns the lowercase equivalent of this string, as a new SharedString.
+    ///
+    /// For example:
+    /// \code
+    ///     auto str = slint::SharedString("Hello");
+    ///     auto str2 = str.to_lowercase(); // creates "hello"
+    /// \endcode
+    const SharedString to_lowercase()
+    {
+        auto out = SharedString();
+        cbindgen_private::slint_shared_string_to_lowercase(&out, this);
+        return out;
+    }
+
+    /// Returns the uppercase equivalent of this string, as a new SharedString.
+    ///
+    /// For example:
+    /// \code
+    ///     auto str = slint::SharedString("Hello");
+    ///     auto str2 = str.to_uppercase(); // creates "HELLO"
+    /// \endcode
+    const SharedString to_uppercase()
```

Comment:
The function needs to be const, not the returned object :)

```suggestion
    SharedString to_lowercase() const
    {
        auto out = SharedString();
        cbindgen_private::slint_shared_string_to_lowercase(&out, this);
        return out;
    }

    /// Returns the uppercase equivalent of this string, as a new SharedString.
    ///
    /// For example:
    /// \code
    ///     auto str = slint::SharedString("Hello");
    ///     auto str2 = str.to_uppercase(); // creates "HELLO"
    /// \endcode
    SharedString to_uppercase() const
```